### DISCUSSION TITLE
Renovate updates dependencies to v999.9.9

### DISCUSSION THREAD

**Discussion by soimon** [renovatebot/renovate#11344] [Request Help] [answered]
### How are you running Renovate?

WhiteSource Renovate hosted app on github.com

### Describe the bug

Renovate recently upgraded all dependencies from the Nx monorepo to an obviously incorrect version `999.9.9`:
| Package | Change | Age | Adoption | Passing | Confidence |
|---|---|---|---|---|---|
| [@nrwl/cli](https://nx.dev) ([source](https://togithub.com/nrwl/nx)) | [`12.7.1` -> `999.9.9`](https://renovatebot.com/diffs/npm/@nrwl%2fcli/12.7.1/999.9.9) | [![age](https://badges.renovateapi.com/packages/npm/@nrwl%2fcli/999.9.9/age-slim)](https://docs.renovatebot.com/merge-confidence/) | [![adoption](https://badges.renovateapi.com/packages/npm/@nrwl%2fcli/999.9.9/adoption-slim)](https://docs.renovatebot.com/merge-confidence/) | [![passing](https://badges.renovateapi.com/packages/npm/@nrwl%2fcli/999.9.9/compatibility-slim/12.7.1)](https://docs.renovatebot.com/merge-confidence/) | [![confidence](https://badges.renovateapi.com/packages/npm/@nrwl%2fcli/999.9.9/confidence-slim/12.7.1)](https://docs.renovatebot.com/merge-confidence/) |
| [@nrwl/eslint-plugin-nx](https://nx.dev) ([source](https://togithub.com/nrwl/nx)) | [`12.7.1` -> `999.9.9`](https://renovatebot.com/diffs/npm/@nrwl%2feslint-plugin-nx/12.7.1/999.9.9) | [![age](https://badges.renovateapi.com/packages/npm/@nrwl%2feslint-plugin-nx/999.9.9/age-slim)](https://docs.renovatebot.com/merge-confidence/) | [![adoption](https://badges.renovateapi.com/packages/npm/@nrwl%2feslint-plugin-nx/999.9.9/adoption-slim)](https://docs.renovatebot.com/merge-confidence/) | [![passing](https://badges.renovateapi.com/packages/npm/@nrwl%2feslint-plugin-nx/999.9.9/compatibility-slim/12.7.1)](https://docs.renovatebot.com/merge-confidence/) | [![confidence](https://badges.renovateapi.com/packages/npm/@nrwl%2feslint-plugin-nx/999.9.9/confidence-slim/12.7.1)](https://docs.renovatebot.com/merge-confidence/) |
| [@nrwl/jest](https://nx.dev) ([source](https://togithub.com/nrwl/nx)) | [`12.7.1` -> `999.9.9`](https://renovatebot.com/diffs/npm/@nrwl%2fjest/12.7.1/999.9.9) | [![age](https://badges.renovateapi.com/packages/npm/@nrwl%2fjest/999.9.9/age-slim)](https://docs.renovatebot.com/merge-confidence/) | [![adoption](https://badges.renovateapi.com/packages/npm/@nrwl%2fjest/999.9.9/adoption-slim)](https://docs.renovatebot.com/merge-confidence/) | [![passing](https://badges.renovateapi.com/packages/npm/@nrwl%2fjest/999.9.9/compatibility-slim/12.7.1)](https://docs.renovatebot.com/merge-confidence/) | [![confidence](https://badges.renovateapi.com/packages/npm/@nrwl%2fjest/999.9.9/confidence-slim/12.7.1)](https://docs.renovatebot.com/merge-confidence/) |
| [@nrwl/linter](https://nx.dev) ([source](https://togithub.com/nrwl/nx)) | [`12.7.1` -> `999.9.9`](https://renovatebot.com/diffs/npm/@nrwl%2flinter/12.7.1/999.9.9) | [![age](https://badges.renovateapi.com/packages/npm/@nrwl%2flinter/999.9.9/age-slim)](https://docs.renovatebot.com/merge-confidence/) | [![adoption](https://badges.renovateapi.com/packages/npm/@nrwl%2flinter/999.9.9/adoption-slim)](https://docs.renovatebot.com/merge-confidence/) | [![passing](https://badges.renovateapi.com/packages/npm/@nrwl%2flinter/999.9.9/compatibility-slim/12.7.1)](https://docs.renovatebot.com/merge-confidence/) | [![confidence](https://badges.renovateapi.com/packages/npm/@nrwl%2flinter/999.9.9/confidence-slim/12.7.1)](https://docs.renovatebot.com/merge-confidence/) |
| [@nrwl/node](https://nx.dev) ([source](https://togithub.com/nrwl/nx)) | [`12.7.1` -> `999.9.9`](https://renovatebot.com/diffs/npm/@nrwl%2fnode/12.7.1/999.9.9) | [![age](https://badges.renovateapi.com/packages/npm/@nrwl%2fnode/999.9.9/age-slim)](https://docs.renovatebot.com/merge-confidence/) | [![adoption](https://badges.renovateapi.com/packages/npm/@nrwl%2fnode/999.9.9/adoption-slim)](https://docs.renovatebot.com/merge-confidence/) | [![passing](https://badges.renovateapi.com/packages/npm/@nrwl%2fnode/999.9.9/compatibility-slim/12.7.1)](https://docs.renovatebot.com/merge-confidence/) | [![confidence](https://badges.renovateapi.com/packages/npm/@nrwl%2fnode/999.9.9/confidence-slim/12.7.1)](https://docs.renovatebot.com/merge-confidence/) |
| [@nrwl/tao](https://nx.dev) ([source](https://togithub.com/nrwl/nx)) | [`12.7.1` -> `999.9.9`](https://renovatebot.com/diffs/npm/@nrwl%2ftao/12.7.1/999.9.9) | [![age](https://badges.renovateapi.com/packages/npm/@nrwl%2ftao/999.9.9/age-slim)](https://docs.renovatebot.com/merge-confidence/) | [![adoption](https://badges.renovateapi.com/packages/npm/@nrwl%2ftao/999.9.9/adoption-slim)](https://docs.renovatebot.com/merge-confidence/) | [![passing](https://badges.renovateapi.com/packages/npm/@nrwl%2ftao/999.9.9/compatibility-slim/12.7.1)](https://docs.renovatebot.com/merge-confidence/) | [![confidence](https://badges.renovateapi.com/packages/npm/@nrwl%2ftao/999.9.9/confidence-slim/12.7.1)](https://docs.renovatebot.com/merge-confidence/) |
| [@nrwl/workspace](https://nx.dev) ([source](https://togithub.com/nrwl/nx)) | [`12.7.1` -> `999.9.9`](https://renovatebot.com/diffs/npm/@nrwl%2fworkspace/12.7.1/999.9.9) | [![age](https://badges.renovateapi.com/packages/npm/@nrwl%2fworkspace/999.9.9/age-slim)](https://docs.renovatebot.com/merge-confidence/) | [![adoption](https://badges.renovateapi.com/packages/npm/@nrwl%2fworkspace/999.9.9/adoption-slim)](https://docs.renovatebot.com/merge-confidence/) | [![passing](https://badges.renovateapi.com/packages/npm/@nrwl%2fworkspace/999.9.9/compatibility-slim/12.7.1)](https://docs.renovatebot.com/merge-confidence/) | [![confidence](https://badges.renovateapi.com/packages/npm/@nrwl%2fworkspace/999.9.9/confidence-slim/12.7.1)](https://docs.renovatebot.com/merge-confidence/) |

My CI setup somehow let this merge (the command `yarn` did not complain about this weird version number, merely warning that `@nrwl/eslint-plugin-nx@999.9.9" has incorrect peer dependency "@typescript-eslint/parser@~4.28.3"`)
A few days later the bot tried to bump the versions even higher to version `9999.0.0`:

| Package | Change | Age | Adoption | Passing | Confidence |
|---|---|---|---|---|---|
| [@nrwl/cli](https://nx.dev) ([source](https://togithub.com/nrwl/nx)) | [`999.9.9` -> `9999.0.0`](https://renovatebot.com/diffs/npm/@nrwl%2fcli/999.9.9/9999.0.0) | [![age](https://badges.renovateapi.com/packages/npm/@nrwl%2fcli/9999.0.0/age-slim)](https://docs.renovatebot.com/merge-confidence/) | [![adoption](https://badges.renovateapi.com/packages/npm/@nrwl%2fcli/9999.0.0/adoption-slim)](https://docs.renovatebot.com/merge-confidence/) | [![passing](https://badges.renovateapi.com/packages/npm/@nrwl%2fcli/9999.0.0/compatibility-slim/999.9.9)](https://docs.renovatebot.com/merge-confidence/) | [![confidence](https://badges.renovateapi.com/packages/npm/@nrwl%2fcli/9999.0.0/confidence-slim/999.9.9)](https://docs.renovatebot.com/merge-confidence/) |
| [@nrwl/jest](https://nx.dev) ([source](https://togithub.com/nrwl/nx)) | [`999.9.9` -> `9999.0.0`](https://renovatebot.com/diffs/npm/@nrwl%2fjest/999.9.9/9999.0.0) | [![age](https://badges.renovateapi.com/packages/npm/@nrwl%2fjest/9999.0.0/age-slim)](https://docs.renovatebot.com/merge-confidence/) | [![adoption](https://badges.renovateapi.com/packages/npm/@nrwl%2fjest/9999.0.0/adoption-slim)](https://docs.renovatebot.com/merge-confidence/) | [![passing](https://badges.renovateapi.com/packages/npm/@nrwl%2fjest/9999.0.0/compatibility-slim/999.9.9)](https://docs.renovatebot.com/merge-confidence/) | [![confidence](https://badges.renovateapi.com/packages/npm/@nrwl%2fjest/9999.0.0/confidence-slim/999.9.9)](https://docs.renovatebot.com/merge-confidence/) |
| [@nrwl/linter](https://nx.dev) ([source](https://togithub.com/nrwl/nx)) | [`999.9.9` -> `9999.0.0`](https://renovatebot.com/diffs/npm/@nrwl%2flinter/999.9.9/9999.0.0) | [![age](https://badges.renovateapi.com/packages/npm/@nrwl%2flinter/9999.0.0/age-slim)](https://docs.renovatebot.com/merge-confidence/) | [![adoption](https://badges.renovateapi.com/packages/npm/@nrwl%2flinter/9999.0.0/adoption-slim)](https://docs.renovatebot.com/merge-confidence/) | [![passing](https://badges.renovateapi.com/packages/npm/@nrwl%2flinter/9999.0.0/compatibility-slim/999.9.9)](https://docs.renovatebot.com/merge-confidence/) | [![confidence](https://badges.renovateapi.com/packages/npm/@nrwl%2flinter/9999.0.0/confidence-slim/999.9.9)](https://docs.renovatebot.com/merge-confidence/) |
| [@nrwl/node](https://nx.dev) ([source](https://togithub.com/nrwl/nx)) | [`999.9.9` -> `9999.0.0`](https://renovatebot.com/diffs/npm/@nrwl%2fnode/999.9.9/9999.0.0) | [![age](https://badges.renovateapi.com/packages/npm/@nrwl%2fnode/9999.0.0/age-slim)](https://docs.renovatebot.com/merge-confidence/) | [![adoption](https://badges.renovateapi.com/packages/npm/@nrwl%2fnode/9999.0.0/adoption-slim)](https://docs.renovatebot.com/merge-confidence/) | [![passing](https://badges.renovateapi.com/packages/npm/@nrwl%2fnode/9999.0.0/compatibility-slim/999.9.9)](https://docs.renovatebot.com/merge-confidence/) | [![confidence](https://badges.renovateapi.com/packages/npm/@nrwl%2fnode/9999.0.0/confidence-slim/999.9.9)](https://docs.renovatebot.com/merge-confidence/) |
| [@nrwl/tao](https://nx.dev) ([source](https://togithub.com/nrwl/nx)) | [`999.9.9` -> `9999.0.0`](https://renovatebot.com/diffs/npm/@nrwl%2ftao/999.9.9/9999.0.0) | [![age](https://badges.renovateapi.com/packages/npm/@nrwl%2ftao/9999.0.0/age-slim)](https://docs.renovatebot.com/merge-confidence/) | [![adoption](https://badges.renovateapi.com/packages/npm/@nrwl%2ftao/9999.0.0/adoption-slim)](https://docs.renovatebot.com/merge-confidence/) | [![passing](https://badges.renovateapi.com/packages/npm/@nrwl%2ftao/9999.0.0/compatibility-slim/999.9.9)](https://docs.renovatebot.com/merge-confidence/) | [![confidence](https://badges.renovateapi.com/packages/npm/@nrwl%2ftao/9999.0.0/confidence-slim/999.9.9)](https://docs.renovatebot.com/merge-confidence/) |
| [@nrwl/workspace](https://nx.dev) ([source](https://togithub.com/nrwl/nx)) | [`999.9.9` -> `9999.0.0`](https://renovatebot.com/diffs/npm/@nrwl%2fworkspace/999.9.9/9999.0.0) | [![age](https://badges.renovateapi.com/packages/npm/@nrwl%2fworkspace/9999.0.0/age-slim)](https://docs.renovatebot.com/merge-confidence/) | [![adoption](https://badges.renovateapi.com/packages/npm/@nrwl%2fworkspace/9999.0.0/adoption-slim)](https://docs.renovatebot.com/merge-confidence/) | [![passing](https://badges.renovateapi.com/packages/npm/@nrwl%2fworkspace/9999.0.0/compatibility-slim/999.9.9)](https://docs.renovatebot.com/merge-confidence/) | [![confidence](https://badges.renovateapi.com/packages/npm/@nrwl%2fworkspace/9999.0.0/confidence-slim/999.9.9)](https://docs.renovatebot.com/merge-confidence/) |

<details><summary>This is my `renovate.json`.</summary>

```
{
	"extends": ["config:base"],
	"labels": ["dependencies"],
	"commitMessagePrefix": "⬆",
	"commitMessageAction": "Upgrade",
	"packageRules": [
		{
			"matchPackagePatterns": ["*"],
			"matchUpdateTypes": ["minor", "patch"],
			"matchCurrentVersion": "!/^0/",
			"groupName": "all non-major dependencies",
			"automerge": true
		},
		{
			"matchPackagePrefixes": ["@types/"],
			"groupName": "definitelyTyped definitions",
			"automerge": true
		},
		{
			"matchPackagePatterns": ["jest"],
			"matchDepTypes": ["devDependencies"],
			"groupName": "Jest",
			"automerge": true
		},
		{
			"matchDepTypes": ["devDependencies"],
			"matchPackagePatterns": ["lint", "prettier"],
			"groupName": "linters and formatters",
			"automerge": true
		},
		{
			"matchPackagePrefixes": ["@nrwl/"],
			"matchDepTypes": ["devDependencies"],
			"groupName": "Nx",
			"automerge": true
		},
		{
			"updateTypes": ["pin"],
			"commitMessagePrefix": "📌",
			"commitMessageAction": "Pin"
		},
		{
			"updateTypes": ["rollback"],
			"commitMessagePrefix": "⬇️",
			"commitMessageAction": "Downgrade"
		}
	],
	"lockFileMaintenance": {
		"enabled": true,
		"automerge": true
	}
}
```
</details>

### Relevant debug logs

<details><summary>Logs</summary>

```
DEBUG: branchifyUpgrades
DEBUG: Using group branchName template
DEBUG: Dependency @nrwl/cli is part of group Nx
DEBUG: Using group branchName template
DEBUG: Dependency @nrwl/eslint-plugin-nx is part of group Nx
DEBUG: Using group branchName template
DEBUG: Dependency @nrwl/jest is part of group Nx
DEBUG: Using group branchName template
DEBUG: Dependency @nrwl/linter is part of group Nx
DEBUG: Using group branchName template
DEBUG: Dependency @nrwl/node is part of group Nx
DEBUG: Using group branchName template
DEBUG: Dependency @nrwl/tao is part of group Nx
DEBUG: Using group branchName template
DEBUG: Dependency @nrwl/workspace is part of group Nx
DEBUG: Using group branchName template
DEBUG: Dependency @typescript-eslint/eslint-plugin is part of group linters and formatters
DEBUG: Using group branchName template
DEBUG: Dependency @typescript-eslint/parser is part of group linters and formatters
DEBUG: 10 flattened updates found: @nrwl/cli, @nrwl/eslint-plugin-nx, @nrwl/jest, @nrwl/linter, @nrwl/node, @nrwl/tao, @nrwl/workspace, @typescript-eslint/eslint-plugin, @typescript-eslint/parser
DEBUG: Returning 3 branch(es)
DEBUG: Fetching changelog: https://github.com/nrwl/nx (12.7.1 -> 999.9.9)
DEBUG: Fetching changelog: https://github.com/nrwl/nx (12.7.1 -> 999.9.9)
DEBUG: Fetching changelog: https://github.com/nrwl/nx (12.7.1 -> 999.9.9)
DEBUG: Fetching changelog: https://github.com/nrwl/nx (12.7.1 -> 999.9.9)
DEBUG: Fetching changelog: https://github.com/nrwl/nx (12.7.1 -> 999.9.9)
DEBUG: Fetching changelog: https://github.com/nrwl/nx (12.7.1 -> 999.9.9)
DEBUG: Fetching changelog: https://github.com/nrwl/nx (12.7.1 -> 999.9.9)
DEBUG: Fetching changelog: https://github.com/typescript-eslint/typescript-eslint (4.29.1 -> 4.29.2)
DEBUG: Fetching changelog: https://github.com/typescript-eslint/typescript-eslint (4.29.1 -> 4.29.2)
DEBUG: config.repoIsOnboarded=true
DEBUG: packageFiles with updates
{
  "config": {
    "npm": [
      {
        "packageFile": "package.json",
        "deps": [
          {
            "depType": "dependencies",
            "depName": "tslib",
            "currentValue": "2.3.1",
            "datasource": "npm",
            "prettyDepType": "dependency",
            "lockedVersion": "2.3.1",
            "depIndex": 0,
            "warnings": [],
            "versioning": "npm",
            "sourceUrl": "https://github.com/Microsoft/tslib",
            "homepage": "https://www.typescriptlang.org/",
            "currentVersion": "2.3.1",
            "fixedVersion": "2.3.1",
            "updates": []
          },
          {
            "depType": "devDependencies",
            "depName": "@nrwl/cli",
            "currentValue": "12.7.1",
            "datasource": "npm",
            "prettyDepType": "devDependency",
            "lockedVersion": "12.7.1",
            "depIndex": 1,
            "warnings": [],
            "versioning": "npm",
            "sourceUrl": "https://github.com/nrwl/nx",
            "homepage": "https://nx.dev",
            "currentVersion": "12.7.1",
            "fixedVersion": "12.7.1",
            "updates": [
              {
                "bucket": "major",
                "newVersion": "999.9.9",
                "newValue": "999.9.9",
                "releaseTimestamp": "2021-08-16T18:48:32.441Z",
                "newMajor": 999,
                "newMinor": 9,
                "updateType": "major",
                "branchName": "renovate/major-nx"
              }
            ],
            "isSingleVersion": true
          },
          {
            "depType": "devDependencies",
            "depName": "@nrwl/eslint-plugin-nx",
            "currentValue": "12.7.1",
            "datasource": "npm",
            "prettyDepType": "devDependency",
            "lockedVersion": "12.7.1",
            "depIndex": 2,
            "warnings": [],
            "versioning": "npm",
            "sourceUrl": "https://github.com/nrwl/nx",
            "homepage": "https://nx.dev",
            "currentVersion": "12.7.1",
            "fixedVersion": "12.7.1",
            "updates": [
              {
                "bucket": "major",
                "newVersion": "999.9.9",
                "newValue": "999.9.9",
                "releaseTimestamp": "2021-08-16T18:48:55.779Z",
                "newMajor": 999,
                "newMinor": 9,
                "updateType": "major",
                "branchName": "renovate/major-nx"
              }
            ],
            "isSingleVersion": true
          },
          {
            "depType": "devDependencies",
            "depName": "@nrwl/jest",
            "currentValue": "12.7.1",
            "datasource": "npm",
            "prettyDepType": "devDependency",
            "lockedVersion": "12.7.1",
            "depIndex": 3,
            "warnings": [],
            "versioning": "npm",
            "sourceUrl": "https://github.com/nrwl/nx",
            "homepage": "https://nx.dev",
            "currentVersion": "12.7.1",
            "fixedVersion": "12.7.1",
            "updates": [
              {
                "bucket": "major",
                "newVersion": "999.9.9",
                "newValue": "999.9.9",
                "releaseTimestamp": "2021-08-16T18:49:08.309Z",
                "newMajor": 999,
                "newMinor": 9,
                "updateType": "major",
                "branchName": "renovate/major-nx"
              }
            ],
            "isSingleVersion": true
          },
          {
            "depType": "devDependencies",
            "depName": "@nrwl/linter",
            "currentValue": "12.7.1",
            "datasource": "npm",
            "prettyDepType": "devDependency",
            "lockedVersion": "12.7.1",
            "depIndex": 4,
            "warnings": [],
            "versioning": "npm",
            "sourceUrl": "https://github.com/nrwl/nx",
            "homepage": "https://nx.dev",
            "currentVersion": "12.7.1",
            "fixedVersion": "12.7.1",
            "updates": [
              {
                "bucket": "major",
                "newVersion": "999.9.9",
                "newValue": "999.9.9",
                "releaseTimestamp": "2021-08-16T18:49:14.741Z",
                "newMajor": 999,
                "newMinor": 9,
                "updateType": "major",
                "branchName": "renovate/major-nx"
              }
            ],
            "isSingleVersion": true
          },
          {
            "depType": "devDependencies",
            "depName": "@nrwl/node",
            "currentValue": "12.7.1",
            "datasource": "npm",
            "prettyDepType": "devDependency",
            "lockedVersion": "12.7.1",
            "depIndex": 5,
            "warnings": [],
            "versioning": "npm",
            "sourceUrl": "https://github.com/nrwl/nx",
            "homepage": "https://nx.dev",
            "currentVersion": "12.7.1",
            "fixedVersion": "12.7.1",
            "updates": [
              {
                "bucket": "major",
                "newVersion": "999.9.9",
                "newValue": "999.9.9",
                "releaseTimestamp": "2021-08-16T18:49:28.754Z",
                "newMajor": 999,
                "newMinor": 9,
                "updateType": "major",
                "branchName": "renovate/major-nx"
              }
            ],
            "isSingleVersion": true
          },
          {
            "depType": "devDependencies",
            "depName": "@nrwl/tao",
            "currentValue": "12.7.1",
            "datasource": "npm",
            "prettyDepType": "devDependency",
            "lockedVersion": "12.7.1",
            "depIndex": 6,
            "warnings": [],
            "versioning": "npm",
            "sourceUrl": "https://github.com/nrwl/nx",
            "homepage": "https://nx.dev",
            "currentVersion": "12.7.1",
            "fixedVersion": "12.7.1",
            "updates": [
              {
                "bucket": "major",
                "newVersion": "999.9.9",
                "newValue": "999.9.9",
                "releaseTimestamp": "2021-08-16T18:49:50.806Z",
                "newMajor": 999,
                "newMinor": 9,
                "updateType": "major",
                "branchName": "renovate/major-nx"
              }
            ],
            "isSingleVersion": true
          },
          {
            "depType": "devDependencies",
            "depName": "@nrwl/workspace",
            "currentValue": "12.7.1",
            "datasource": "npm",
            "prettyDepType": "devDependency",
            "lockedVersion": "12.7.1",
            "depIndex": 7,
            "warnings": [],
            "versioning": "npm",
            "sourceUrl": "https://github.com/nrwl/nx",
            "homepage": "https://nx.dev",
            "currentVersion": "12.7.1",
            "fixedVersion": "12.7.1",
            "updates": [
              {
                "bucket": "major",
                "newVersion": "999.9.9",
                "newValue": "999.9.9",
                "releaseTimestamp": "2021-08-16T18:50:01.031Z",
                "newMajor": 999,
                "newMinor": 9,
                "updateType": "major",
                "branchName": "renovate/major-nx"
              }
            ],
            "isSingleVersion": true
          },
          {
            "depType": "devDependencies",
            "depName": "@types/jest",
            "currentValue": "27.0.1",
            "datasource": "npm",
            "prettyDepType": "devDependency",
            "lockedVersion": "27.0.1",
            "depIndex": 8,
            "warnings": [],
            "versioning": "npm",
            "sourceUrl": "https://github.com/DefinitelyTyped/DefinitelyTyped",
            "sourceDirectory": "types/jest",
            "currentVersion": "27.0.1",
            "fixedVersion": "27.0.1",
            "updates": []
          },
          {
            "depType": "devDependencies",
            "depName": "@types/node",
            "currentValue": "16.4.14",
            "datasource": "npm",
            "prettyDepType": "devDependency",
            "lockedVersion": "16.4.14",
            "depIndex": 9,
            "warnings": [],
            "versioning": "node",
            "sourceUrl": "https://github.com/DefinitelyTyped/DefinitelyTyped",
            "sourceDirectory": "types/node",
            "currentVersion": "16.4.14",
            "fixedVersion": "16.4.14",
            "updates": []
          },
          {
            "depType": "devDependencies",
            "depName": "@typescript-eslint/eslint-plugin",
            "currentValue": "4.29.1",
            "datasource": "npm",
            "prettyDepType": "devDependency",
            "lockedVersion": "4.29.1",
            "depIndex": 10,
            "warnings": [],
            "versioning": "npm",
            "sourceUrl": "https://github.com/typescript-eslint/typescript-eslint",
            "sourceDirectory": "packages/eslint-plugin",
            "currentVersion": "4.29.1",
            "fixedVersion": "4.29.1",
            "isSingleVersion": true,
            "updates": [
              {
                "bucket": "non-major",
                "newVersion": "4.29.2",
                "newValue": "4.29.2",
                "releaseTimestamp": "2021-08-16T17:04:17.067Z",
                "newMajor": 4,
                "newMinor": 29,
                "updateType": "patch",
                "branchName": "renovate/linters-and-formatters"
              }
            ]
          },
          {
            "depType": "devDependencies",
            "depName": "@typescript-eslint/parser",
            "currentValue": "4.29.1",
            "datasource": "npm",
            "prettyDepType": "devDependency",
            "lockedVersion": "4.29.1",
            "depIndex": 11,
            "warnings": [],
            "versioning": "npm",
            "sourceUrl": "https://github.com/typescript-eslint/typescript-eslint",
            "sourceDirectory": "packages/parser",
            "currentVersion": "4.29.1",
            "fixedVersion": "4.29.1",
            "isSingleVersion": true,
            "updates": [
              {
                "bucket": "non-major",
                "newVersion": "4.29.2",
                "newValue": "4.29.2",
                "releaseTimestamp": "2021-08-16T17:04:07.967Z",
                "newMajor": 4,
                "newMinor": 29,
                "updateType": "patch",
                "branchName": "renovate/linters-and-formatters"
              }
            ]
          },
          {
            "depType": "devDependencies",
            "depName": "dotenv",
            "currentValue": "10.0.0",
            "datasource": "npm",
            "prettyDepType": "devDependency",
            "lockedVersion": "10.0.0",
            "depIndex": 12,
            "warnings": [],
            "versioning": "npm",
            "sourceUrl": "https://github.com/motdotla/dotenv",
            "currentVersion": "10.0.0",
            "fixedVersion": "10.0.0",
            "updates": []
          },
          {
            "depType": "devDependencies",
            "depName": "eslint",
            "currentValue": "7.32.0",
            "datasource": "npm",
            "prettyDepType": "devDependency",
            "lockedVersion": "7.32.0",
            "depIndex": 13,
            "warnings": [],
            "versioning": "npm",
            "sourceUrl": "https://github.com/eslint/eslint",
            "homepage": "https://eslint.org",
            "currentVersion": "7.32.0",
            "fixedVersion": "7.32.0",
            "updates": []
          },
          {
            "depType": "devDependencies",
            "depName": "eslint-config-prettier",
            "currentValue": "8.3.0",
            "datasource": "npm",
            "prettyDepType": "devDependency",
            "lockedVersion": "8.3.0",
            "depIndex": 14,
            "warnings": [],
            "versioning": "npm",
            "sourceUrl": "https://github.com/prettier/eslint-config-prettier",
            "currentVersion": "8.3.0",
            "fixedVersion": "8.3.0",
            "updates": []
          },
          {
            "depType": "devDependencies",
            "depName": "jest",
            "currentValue": "27.0.6",
            "datasource": "npm",
            "prettyDepType": "devDependency",
            "lockedVersion": "27.0.6",
            "depIndex": 15,
            "warnings": [],
            "versioning": "npm",
            "sourceUrl": "https://github.com/facebook/jest",
            "homepage": "https://jestjs.io/",
            "currentVersion": "27.0.6",
            "fixedVersion": "27.0.6",
            "updates": []
          },
          {
            "depType": "devDependencies",
            "depName": "prettier",
            "currentValue": "2.3.2",
            "datasource": "npm",
            "prettyDepType": "devDependency",
            "lockedVersion": "2.3.2",
            "depIndex": 16,
            "warnings": [],
            "versioning": "npm",
            "sourceUrl": "https://github.com/prettier/prettier",
            "homepage": "https://prettier.io",
            "currentVersion": "2.3.2",
            "fixedVersion": "2.3.2",
            "updates": []
          },
          {
            "depType": "devDependencies",
            "depName": "ts-jest",
            "currentValue": "27.0.4",
            "datasource": "npm",
            "prettyDepType": "devDependency",
            "lockedVersion": "27.0.4",
            "depIndex": 17,
            "warnings": [],
            "versioning": "npm",
            "sourceUrl": "https://github.com/kulshekhar/ts-jest",
            "homepage": "https://kulshekhar.github.io/ts-jest",
            "currentVersion": "27.0.4",
            "fixedVersion": "27.0.4",
            "updates": []
          },
          {
            "depType": "devDependencies",
            "depName": "ts-node",
            "currentValue": "10.2.0",
            "datasource": "npm",
            "prettyDepType": "devDependency",
            "lockedVersion": "10.2.0",
            "depIndex": 18,
            "warnings": [],
            "versioning": "npm",
            "sourceUrl": "https://github.com/TypeStrong/ts-node",
            "homepage": "https://typestrong.org/ts-node",
            "currentVersion": "10.2.0",
            "fixedVersion": "10.2.0",
            "updates": []
          },
          {
            "depType": "devDependencies",
            "depName": "typescript",
            "currentValue": "4.3.5",
            "datasource": "npm",
            "prettyDepType": "devDependency",
            "lockedVersion": "4.3.5",
            "depIndex": 19,
            "warnings": [],
            "versioning": "npm",
            "sourceUrl": "https://github.com/Microsoft/TypeScript",
            "homepage": "https://www.typescriptlang.org/",
            "currentVersion": "4.3.5",
            "fixedVersion": "4.3.5",
            "updates": []
          }
        ],
        "packageJsonName": "stagebrain",
        "packageFileVersion": "0.0.0",
        "packageJsonType": "app",
        "yarnLock": "yarn.lock",
        "managerData": {},
        "skipInstalls": true,
        "constraints": {},
        "lockFiles": [
          "yarn.lock"
        ]
      }
    ]
  }
}
```

</details>

### Have you created a minimal reproduction repository?

No reproduction repository

**Comment by viceice** [comment, marked as answer]
This was by design, because nrwl wrongly released version `999.9.9` and later they deprecated that version

![image](https://user-images.githubusercontent.com/1798109/130100254-6dc2441e-a973-4152-b793-e8187bd2c2df.png)
https://www.npmjs.com/package/@nrwl/cli

So your got what you expect if you allow automerging major updates.

You should use `stabilityDays` to defer major updates to certain stabillity.

https://docs.renovatebot.com/configuration-options/#stabilitydays

**Reply by soimon** [reply]
Ah, that makes total sense. I had looked through the version history but somehow overlooked the _show deprecated versions_ checkbox. Thanks for your time and your recommendation.

**Reply by viceice** [reply]
https://github.com/nrwl/nx/issues/6738
